field_level_encryption/field_level_encryption_params.go

```diff
@@ -0,0 +1,51 @@
+package field_level_encryption
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"github.com/mastercard/client-encryption-go/aes_encryption"
+	"github.com/mastercard/client-encryption-go/utils"
+)
+
+type FieldLevelEncryptionParams struct {
+	SymmetricKeySize                int
+	SymmetricKeyType                string
+	IvValue                         string
+	EncryptedKeyValue               string
+	OaepPaddingDigestAlgorithmValue string
+	Config                          *FieldLevelEncryptionConfig
+	SecretKey                       []byte
+	IvParameterSpec                 []byte
+}
+
+func NewFieldLevelEncryptionParams(ivValue string, encryptedKeyValue string, oaepPaddingDigestAlgorithmValue string, config *FieldLevelEncryptionConfig) *FieldLevelEncryptionParams {
+	flep := FieldLevelEncryptionParams{}
+	flep.SymmetricKeySize = 128
+	flep.SymmetricKeyType = "AES"
+
+	flep.IvValue = ivValue
+	flep.EncryptedKeyValue = encryptedKeyValue
+	flep.OaepPaddingDigestAlgorithmValue = oaepPaddingDigestAlgorithmValue
+	flep.Config = config
+
+	return &flep
+}
+
+func Generate(config *FieldLevelEncryptionConfig) *FieldLevelEncryptionParams {
+	//// Generate a random IV
+	ivParameterSpec := aes_encryption.GenerateCEK(16 * 8)
+	ivSpecValue := utils.HexUrlEncode(ivParameterSpec)
```

Comment:
In our other libraries (Java for example), we allow the user to specify the FieldValueEncoding through the config. In your implementation it's hard coded to always use Hex.